listenstore/listenstore/listenstore.py

```diff
@@ -104,21 +110,126 @@ class ListenStore(object):
 
     def __init__(self, conf):
         self.log = logging.getLogger(__name__)
-        self.replication_factor = conf.get("replication_factor", 1)
-        self.keyspace = conf.get("cassandra_keyspace", "listenbrainz")
-        host = conf.get("cassandra_server", "localhost:9092")
+
+    def max_id(self):
+        return int(self.MAX_FUTURE_SECONDS + calendar.timegm(time.gmtime()))
+
+    def format_dict(self, listen):
+        return {'uid': listen.uid,
+                  'id': listen.timestamp,
+                  'date': listen.date,
+                  'artist_msid': uuid.UUID(listen.artist_msid),
+                  'album_msid': uuid.UUID(listen.album_msid) if listen.album_msid is not None else None,
+                  'recording_msid': uuid.UUID(listen.recording_msid),
+                  'json': ujson.dumps(listen.data)}
+
+    def fetch_listens(self, uid, from_id=None, to_id=None, limit=None):
+        """ Setup from_id, to_id, and limit for fetching listens
+        """
+        if from_id and to_id:
+            raise ValueError("You cannot specify from_id and to_id at the same time.")
+
+        if from_id:
+            order = ORDER_ASC
+        else:
+            order = ORDER_DESC
+
+        precision = 'month'
+        if from_id is None:
+            from_id = MIN_ID
+        if to_id is None:
+            to_id = self.max_id()
+        return self.fetch_listens_from_database(uid, from_id, to_id, limit, order, precision)
+
+
+class PostgresListenStore(ListenStore):
+    def __init__(self, conf):
+        ListenStore.__init__(self, conf)
+        self.log.info('Connecting to postgresql: %s', conf['sqlalchemy_database_uri'])
+        self.engine = create_engine(conf['sqlalchemy_database_uri'], poolclass=NullPool)
+        self.connection = self.engine.connect()
+        self.connection.execute("SET synchronous_commit TO off;")
```

Comment:
Asynchronous commit is an option that allows transactions to complete(virtually complete) more quickly. Since the no of commits are quite large in each transaction, and the length of each transaction is about 300 listens, so making the commit process async may boost up the actual speed at which the lastfm receives the listens.

I was not sure how to enable this(other than manually changing the postgres config file), but my guess was to switch it on before every transaction.

I really think this might help get a speed up. More details.
https://www.postgresql.org/docs/9.4/static/wal-async-commit.html